### ISSUE TITLE
Error: Building package magnum-extras:x86-windows failed with: BUILD_FAILED

### ISSUE THREAD

**Issue by aimspike** [mosra/magnum-extras#7]
Actually build failed when I installed magnum-extras[ui]:x86-windows

here is the log
```
PS D:\WorkSpace\vcpkg> .\vcpkg.exe install magnum-extras[ui]
The following packages will be built and installed:
    magnum-extras[core,ui]:x86-windows
Starting package 1/1: magnum-extras:x86-windows
Building package magnum-extras[core,ui]:x86-windows...
-- Using cached D:/WorkSpace/vcpkg/downloads/mosra-magnum-extras-v2018.04.tar.gz
-- Using source at D:/WorkSpace/vcpkg/buildtrees/magnum-extras/src/v2018.04-c2a8d6e41e
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_build_cmake.cmake:175 (message):
    Command failed: C:/Program Files/CMake/bin/cmake.exe;--build;.;--config;Debug;--target;install;--;-v;-j1
    Working Directory: D:/WorkSpace/vcpkg/buildtrees/magnum-extras/x86-windows-dbg
    See logs for more information:
      D:\WorkSpace\vcpkg\buildtrees\magnum-extras\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/magnum-extras/portfile.cmake:38 (vcpkg_install_cmake)
  scripts/ports.cmake:71 (include)


Error: Building package magnum-extras:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: magnum-extras:x86-windows
  Vcpkg version: 0.0.113-nohash

Additionally, attach any relevant sections from the log files above.
```

and log in  *D:\WorkSpace\vcpkg\buildtrees\magnum-extras\install-x86-windows-dbg-out.log*

```
D:\WorkSpace\vcpkg\buildtrees\magnum-extras\src\v2018.04-c2a8d6e41e\src\Magnum/Ui/Style.h(482): note: see declaration of 'Magnum::Ui::Implementation::TextShader'
D:\WorkSpace\vcpkg\installed\x86-windows\include\Corrade/Interconnect/Emitter.h(533): error C2338: Size of member function pointer is incorrectly assumed to be smaller than 2*sizeof(void*)
D:\WorkSpace\vcpkg\buildtrees\magnum-extras\src\v2018.04-c2a8d6e41e\src\Magnum\Ui\ValidatedInput.cpp(45): note: see reference to function template instantiation 'Corrade::Interconnect::Connection Corrade::Interconnect::connect<Magnum::Ui::ValidatedInput,Magnum::Ui::Input,Magnum::Ui::ValidatedInput,Magnum::Ui::ValidatedInput,const std::string&>(EmitterObject &,Corrade::Interconnect::Emitter::Signal (__thiscall Magnum::Ui::Input::* )(const std::string &),ReceiverObject &,void (__thiscall Magnum::Ui::ValidatedInput::* )(const std::string &))' being compiled
        with
        [
            EmitterObject=Magnum::Ui::ValidatedInput,
            ReceiverObject=Magnum::Ui::ValidatedInput
        ]
D:\WorkSpace\vcpkg\installed\x86-windows\include\Corrade/Interconnect/Emitter.h(449): warning C4121: 'Corrade::Interconnect::Implementation::MemberConnectionData<ReceiverObject,const std::string &>': alignment of a member was sensitive to packing
        with
        [
            ReceiverObject=Magnum::Ui::ValidatedInput
        ]
D:\WorkSpace\vcpkg\installed\x86-windows\include\Corrade/Interconnect/Emitter.h(545): note: see reference to class template instantiation 'Corrade::Interconnect::Implementation::MemberConnectionData<ReceiverObject,const std::string &>' being compiled
        with
        [
            ReceiverObject=Magnum::Ui::ValidatedInput
        ]
```

**Comment by mosra** [comment]
Oh, sorry about this, I know about the issue but didn't get to fixing it yet :/ This is a problem that's on 32-bit Windows builds only. Installing a 64-bit version works.

Putting this on my TODO list for the [2018.08](https://github.com/mosra/magnum/issues/265) release.

**Comment by aimspike** [comment]
x64-windows compile succeed but failed on running, with exception

_**Unhandled exception at 0x000007FEC86FC1DF (MagnumUi-d.dll) in magnum-arealights.exe: 0xC0000005: Access violation reading location 0x0000000000000030. occurred**_

and it stop at the fourth line in file _**magnum-examples\src\arealights\AreaLightsExample.cpp**_

```
struct BaseUiPlane: Ui::Plane {
    explicit BaseUiPlane(Ui::UserInterface& ui):
        Ui::Plane{ui, Ui::Snap::Top|Ui::Snap::Bottom|Ui::Snap::Left|Ui::Snap::Right, 0, 16, 128},
here --> metalness{*this, {Ui::Snap::Top|Ui::Snap::Right, WidgetSize}, FloatValidator, "0.5", 5},
        roughness{*this, {Ui::Snap::Bottom, metalness, WidgetSize}, FloatValidator, "0.25", 5},
        f0{*this, {Ui::Snap::Bottom, roughness, WidgetSize}, FloatValidator, "0.25", 5},

        apply{*this, {Ui::Snap::Bottom|Ui::Snap::Right, WidgetSize}, "Apply", Ui::Style::Primary},
        reset{*this, {Ui::Snap::Top, apply, WidgetSize}, "Reset", Ui::Style::Danger}
    {
        Ui::Label{*this, {Ui::Snap::Left, metalness}, "Metalness", Text::Alignment::MiddleRight};
        Ui::Label{*this, {Ui::Snap::Left, roughness}, "Roughness", Text::Alignment::MiddleRight};
        Ui::Label{*this, {Ui::Snap::Left, f0}, "ƒ₀", Text::Alignment::MiddleRight};

        Ui::Label{*this, {Ui::Snap::Bottom|Ui::Snap::Left, WidgetSize},
            "Use WASD + mouse to move, (Shift +) M/R/F to change parameters.",
            Text::Alignment::MiddleLeft};
    }

    Ui::ValidatedInput metalness,
        roughness,
        f0;
    Ui::Button apply,
        reset;
};
```

**Comment by mosra** [comment]
Dang, sorry again :sweat_smile: This issue is present in 2018.04, but is fixed in current `master`. If you would install the `--head` version of all Magnum/Corrade packages, this problem goes away.

I really need to release a stable version ASAP...

**Comment by mosra** [comment]
Fixed with mosra/corrade@b83c116754c92bcf57ccedf113657a75fc11e6f5, will be part of the next release :tada:
